libraries/chain/combined_database.cpp

```diff
@@ -9,9 +9,9 @@
 namespace eosio { namespace chain {
    combined_session::combined_session(chainbase::database& cb_database, eosio::session::undo_stack<rocks_db_type>* undo_stack)
        : kv_undo_stack{ undo_stack } {
+      cb_session = std::make_unique<chainbase::database::session>(cb_database.start_undo_session(true));
```

Comment:
Seems this would be better as a std::optional than a unique_ptr that way you avoid the dynamic memory alloc.

Reply:
Changing cb_session to "std::optional<chainbase::database::session> cb_session" involves multiple changes in combined_database.cpp and combined_database.cpp. Is it OK not to do it at this time?

Reply:
Yep

Reply:
Thanks Kevin for all your help and insist on getting best possible code!